accord-core/src/main/java/accord/impl/SimpleProgressLog.java

```diff
@@ -646,6 +649,8 @@ public void waiting(SafeCommand blockedBy, LocalExecution blockedUntil, Route<?>
 
             // ensure we have a record to work with later; otherwise may think has been truncated
             blockedBy.initialise();
+            if (blockedBy.current().has(blockedUntil.requires))
```

Comment:
We don't know if we're blocked by it until we check its state. All we know is that it's in our dependencies and we haven't determined if we need to wait for it. So, we're blocked by it until we determine otherwise.